kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -86,70 +86,56 @@ trait KotlinModule extends JavaModule { outer =>
 
   protected def kotlinWorkerRef: ModuleRef[KotlinWorkerModule] = ModuleRef(KotlinWorkerModule)
 
-  private[kotlinlib] def kotlinWorkerClasspath = Task {
-    defaultResolver().classpath(Seq(
-      Dep.millProjectModule("mill-kotlinlib-worker-impl")
-    ))
-  }
-
   /**
    * The Java classpath resembling the Kotlin compiler.
    * Default is derived from [[kotlinCompilerIvyDeps]].
    */
   def kotlinCompilerClasspath: T[Seq[PathRef]] = Task {
-    defaultResolver().classpath(kotlinCompilerIvyDeps()) ++
-      kotlinWorkerClasspath()
+    val deps = kotlinCompilerIvyDeps() ++ Seq(
+      Dep.millProjectModule("mill-kotlinlib-worker-impl")
+    )
+    defaultResolver().classpath(deps)
   }
 
   /**
-   * Flag to use the embeddable kotlin compiler.
+   * Flag to enable the use the embeddable kotlin compiler.
    * This can be necessary to avoid classpath conflicts or ensure
    * compatibility to the used set of plugins.
    *
-   * See also https://discuss.kotlinlang.org/t/kotlin-compiler-embeddable-vs-kotlin-compiler/3196
-   */
-  def kotlinCompilerEmbeddable: Task[Boolean] = Task { false }
-
-  /**
-   * The kotlin-compiler dependencies.
-   *
-   * It uses the embeddable version, if [[kotlinCompilerEmbeddable]] is `true`.
-   */
-  def kotlinCompilerDep: T[Seq[Dep]] = Task {
-    if (kotlinCompilerEmbeddable())
-      Seq(ivy"org.jetbrains.kotlin:kotlin-compiler-embeddable:${kotlinCompilerVersion()}")
-    else
-      Seq(ivy"org.jetbrains.kotlin:kotlin-compiler:${kotlinCompilerVersion()}")
-  }
-
-  /**
-   * The kotlin-scripting-compiler dependencies.
+   * The difference between the standard compiler and the embedded compiler is,
+   * that the embedded compiler comes as a dependency-free JAR.
+   * All its dependencies are shaded and thus relocated to different package names.
+   * This also affects the compiler API, since relocated types may surface in the API
+   * but are not compatible to their non-relocated versions. E.g. the plugin's dependencies need to line up with the embeddable compiler's shading, otherwise a [[java.lang.AbstractMethodError]] will be thrown.
```

Comment:
```suggestion
   * but are not compatible to their non-relocated versions. 
   * E.g. the plugin's dependencies need to line up with the embeddable compiler's
   * shading, otherwise a [[java.lang.AbstractMethodError]] will be thrown.
```